cfgmgr/fabricmgr.cpp

```diff
@@ -0,0 +1,119 @@
+
+using namespace std;
+using namespace swss;
+
+FabricMgr::FabricMgr(DBConnector *cfgDb, DBConnector *appDb, const vector<string> &tableNames) :
+        Orch(cfgDb, tableNames),
+        m_cfgFabricMonitorTable(cfgDb, CFG_FABRIC_MONITOR_DATA_TABLE_NAME),
+        m_cfgFabricPortTable(cfgDb, CFG_FABRIC_MONITOR_PORT_TABLE_NAME),
+        m_appFabricMonitorTable(appDb, APP_FABRIC_MONITOR_DATA_TABLE_NAME),
+        m_appFabricPortTable(appDb, APP_FABRIC_MONITOR_PORT_TABLE_NAME)
+{
+}
+
+void FabricMgr::doTask(Consumer &consumer)
+{
+    SWSS_LOG_ENTER();
+
+    auto table = consumer.getTableName();
+
+    auto it = consumer.m_toSync.begin();
+    while (it != consumer.m_toSync.end())
+    {
+        KeyOpFieldsValuesTuple t = it->second;
+
+        string key = kfvKey(t);
+        string op = kfvOp(t);
+
+        if (op == SET_COMMAND)
+        {
+
+            string monErrThreshCrcCells, monErrThreshRxCells;
+            string monPollThreshRecovery, monPollThreshIsolation;
+            string isolateStatus;
+            string alias, lanes;
+            std::vector<FieldValueTuple> field_values;
+            string value;
+
+            for (auto i : kfvFieldsValues(t))
+            {
+                if (fvField(i) == "monErrThreshCrcCells")
+                {
+                    monErrThreshCrcCells = fvValue(i);
+                    writeConfigToAppDb(key, "monErrThreshCrcCells", monErrThreshCrcCells);
+                }
+                else if (fvField(i) == "monErrThreshRxCells")
+                {
+                    monErrThreshRxCells = fvValue(i);
+                    writeConfigToAppDb(key, "monErrThreshRxCells", monErrThreshRxCells);
+                }
+                else if (fvField(i) == "monPollThreshRecovery")
+                {
+                    monPollThreshRecovery = fvValue(i);
+                    writeConfigToAppDb(key, "monPollThreshRecovery", monPollThreshRecovery);
+                }
+                else if (fvField(i) == "monPollThreshIsolation")
+                {
+                    monPollThreshIsolation = fvValue(i);
+                    writeConfigToAppDb(key, "monPollThreshIsolation", monPollThreshIsolation);
+                }
+                else if (fvField(i) == "alias")
+                {
+                    alias = fvValue(i);
+                    writeConfigToAppDb(key, "alias", alias);
+                }
+                else if (fvField(i) == "lanes")
+                {
+                    lanes = fvValue(i);
+                    writeConfigToAppDb(key, "lanes", lanes);
+                }
+                else if (fvField(i) == "isolateStatus")
+                {
+                    isolateStatus = fvValue(i);
+                    writeConfigToAppDb(key, "isolateStatus", isolateStatus);
+                }
+                else
+                {
+                    field_values.emplace_back(i);
+                }
+            }
+
+            for (auto &entry : field_values)
```

Comment:
why do we need this?

Reply:
> why do we need this?

This is needed for adding the fields in to the appl_db. other mgr , like portmgr , also do the same thing. thank you